x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -535,14 +536,15 @@ protected List<BucketExtractor> initBucketExtractors(SearchResponse response) {
             List<QueryContainer.FieldInfo> refs = query.fields();
 
             List<BucketExtractor> exts = new ArrayList<>(refs.size());
-            ConstantExtractor totalCount = new ConstantExtractor(response.getHits().getTotalHits().value);
+            TotalHits totalHits = response.getHits().getTotalHits();
+            ConstantExtractor totalCount = new ConstantExtractor(totalHits == null ? -1 : totalHits.value);
```

Comment:
Since -1 is not a valid value (despite showing up in the queryDSL), +1 from to encapsulate this into a dedicated ConstantExtractor-like that throws an exception if used.
This keeps the code clean and passes the circuitbreaker around which will trip instead of returning -1.

Since the topic has already been discussed, I'm approving the ticket from now before the changes are applied.